src/components/components/EditBenefitForm.js

```diff
@@ -0,0 +1,151 @@
+import React from "react";
+import Form from "react-bootstrap/Form";
+import Button from "react-bootstrap/Button";
+import apiBeneficio from "../../services/apiBeneficio";
+import { withRouter , Redirect } from "react-router-dom";
+//import {} from "react-router"
+import { Row } from "react-bootstrap";
+
+
+
+class EditBenefitForm extends React.Component {
+
+    constructor(props){
+      super(props);
+      this.state = {
+        title: "",
+        description: "",
+        price:  null,
+        redeem_way: "" ,
+        quantity: null
+      };
+
+      this.id = this.props.match.params.benefitId;
+
+      // this.history = useHistory();
+  
+      this.handleChange = this.handleChange.bind(this);
+      this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    async getBenefits(){
+      const response = await apiBeneficio.get("benefits/"+this.id);
+      this.setState(response.data);
+    }
+
+    componentDidMount(){
+      this.getBenefits();
+    }
+  
+    handleChange(event){
+      switch(event.target.id){
+        case  "title":
+          this.setState({title: event.target.value});
+          break;
+        case "description":
+          this.setState({description: event.target.value});
+          break;
+        case "price":
+          this.setState({price: event.target.value});
```

Comment:
Codacy found an issue: [Method 'handleChange' has a complexity of 6. Maximum allowed is 4.](https://app.codacy.com/gh/fga-eps-mds/2020.1-VC_Gestor-FrontEnd/pullRequest?prid=6353377)